### ISSUE TITLE
Current master is not backwards compatible with old hdf5 files

### ISSUE THREAD

**Issue by plaplant** [HERA-Team/hera_pspec#194]
It seems that the current version of master is no longer backwards compatible with old files. I am not sure when the transition happened, but there are no clues to the user what is wrong or how to read in older files. Trying to read a pspec object generated with an older version of the code (located at `/lustre/aoc/projects/hera/plaplant/HERA19Golden/CalibratedData/LSThrs_10.5_23.0/FREQrng_530_730/zen.2457548.16700_66112.pI.HH.uvcRPCS.uvh5.UVPavg`) yields this traceback:

```
In [14]: uvp.read_hdf5("zen.2457755.58922_11048.pI.HH.uvCP.uvh5.UVPavg")
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-14-ee8853d6d1eb> in <module>
----> 1 uvp.read_hdf5("zen.2457755.58922_11048.pI.HH.uvCP.uvh5.UVPavg")

/lustre/aoc/projects/hera/plaplant/anaconda2/envs/py3/lib/python3.6/site-packages/hera_pspec-0.1.0-py3.6.egg/hera_pspec/uvpspec.py in read_hdf5(self, filepath, just_meta, spws, bls,
 blpairs, times, lsts, polpairs, only_pairs_in_bls)
   1221             self.read_from_group(f, just_meta=just_meta, spws=spws, bls=bls,
   1222                                  times=times, lsts=lsts, polpairs=polpairs,
-> 1223                                  only_pairs_in_bls=only_pairs_in_bls)
   1224
   1225

/lustre/aoc/projects/hera/plaplant/anaconda2/envs/py3/lib/python3.6/site-packages/hera_pspec-0.1.0-py3.6.egg/hera_pspec/uvpspec.py in read_from_group(self, grp, just_meta, spws, bls
, blpairs, times, lsts, polpairs, only_pairs_in_bls)
   1141         assert 'pspec_type' in grp.attrs, "This object is not a UVPSpec object"
   1142         assert grp.attrs['pspec_type'] == 'UVPSpec', \
-> 1143             "This object is not a UVPSpec object"
   1144
   1145         # Clear all data in the current object

AssertionError: This object is not a UVPSpec object

In [15]: uvp = hera_pspec.UVPSpec()

In [16]: uvp.read_hdf5("zen.2457755.58922_11048.pI.HH.uvCP.uvh5.UVPavg")
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-16-ee8853d6d1eb> in <module>
----> 1 uvp.read_hdf5("zen.2457755.58922_11048.pI.HH.uvCP.uvh5.UVPavg")

/lustre/aoc/projects/hera/plaplant/anaconda2/envs/py3/lib/python3.6/site-packages/hera_pspec-0.1.0-py3.6.egg/hera_pspec/uvpspec.py in read_hdf5(self, filepath, just_meta, spws, bls,
 blpairs, times, lsts, polpairs, only_pairs_in_bls)
   1221             self.read_from_group(f, just_meta=just_meta, spws=spws, bls=bls,
   1222                                  times=times, lsts=lsts, polpairs=polpairs,
-> 1223                                  only_pairs_in_bls=only_pairs_in_bls)
   1224
   1225

/lustre/aoc/projects/hera/plaplant/anaconda2/envs/py3/lib/python3.6/site-packages/hera_pspec-0.1.0-py3.6.egg/hera_pspec/uvpspec.py in read_from_group(self, grp, just_meta, spws, bls
, blpairs, times, lsts, polpairs, only_pairs_in_bls)
   1141         assert 'pspec_type' in grp.attrs, "This object is not a UVPSpec object"
   1142         assert grp.attrs['pspec_type'] == 'UVPSpec', \
-> 1143             "This object is not a UVPSpec object"
   1144
   1145         # Clear all data in the current object

AssertionError: This object is not a UVPSpec object
```
It seems that there are some attributes that are now being saved to the pspec objects that makes older files appear invalid. In the interest of allowing for the validation of current data with existing or new analysis methods at some date in the future, there needs to be some way to read in old files and save them in the latest format. Simply raising an error with no obvious path forward is not an acceptable solution.

In future development, if there are breaking changes to how the files are saved on disk, functionality to read in older files and save them in the new format must be built alongside those changes.

**Comment by philbull** [comment]
Yep, this is due to the change from storing polarisations to polarisation pairs, which broke the API. This should be a straightforward fix -- I'll have a go at it.

**Comment by philbull** [comment]
On closer inspection, this is actually a Python 3/HDF5 issue! HDF5 is returning strings as `bytes`, and there is a string comparison in `UVPSpec` that is failing as a result.

**Comment by plaplant** [comment]
Fixed by #195.
